activesupport/lib/active_support/callbacks.rb

```diff
@@ -826,6 +754,30 @@ def get_callbacks(name)
       def set_callbacks(name, callbacks)
         send "_#{name}_callbacks=", callbacks
       end
+
+      def default_false_terminator
+        Proc.new do |target, result_lambda|
+          terminate = true
+          catch(:abort) do
+            result = result_lambda.call if result_lambda.is_a?(Proc)
+            if CallbackChain.halt_and_display_warning_on_return_false && result == false
```

Comment:
Ah, now I get it. Thanks for the explanation :+1:

It would still feel weird to have a setting on Active Support that then only mentions Active Record and Active Model.

There's some things we could do. Rename the config to `Rails.application.config.callbacks.halt_and_display_warning_on_return_false` and bump these specifics to the upgrading guide.

It also sounds like using Active Support callbacks directly is an edge case, so in our deprecation message it's fine to treat AR and AM callbacks as all the callbacks IMO.

Unless we already have them, we should add some instructions to support `throw(:abort)` for the libraries that define their own callbacks with AS.

What do you think, @claudiob @repinel? :smile: